File: scripts/Single Element in a Sorted Array/Single Element in a Sorted Array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        
        int left = 0;
        int right = nums.length-1;
        
        while(left < right){
            
            int mid = (left+right)/2;
            
            if(nums[mid] == nums[mid+1]){
                mid = mid - 1;                   //two same elements should be in same half
            }
            
            if((mid-left+1)%2 != 0){
                right = mid;                     // checking the length of left half. 
                                                //If its is odd then update ur right pointer to mid
            }else{
                left = mid + 1;                  // else your right half will be odd 
                                                //then update your left pointer to mid+1

            }
        }
        
        return nums[left];                  //left pointer will have the answer at last 
    }
}

File: algorithms/S/Single Element in a Sorted Array/Single Element in a Sorted Array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        return self.b_search(nums)[0]
    
    def b_search(self, nums):
        if len(nums) == 1:
            return nums
        mid = len(nums)//2
        a = nums[:mid]
        b = nums[mid:]
		
		# check if last & first element of the two sub lists are same
        if a[-1] == b[0]:
            a = a[:-1]
            b = b[1:]
		
		# ignore the sub list with even number of elements
        if len(a)%2:
            return self.b_search(a)
        else:
            return self.b_search(b)
